runtime/src/main/java/com/dylibso/chicory/runtime/ExecutionListener.java

```diff
@@ -0,0 +1,8 @@
+package com.dylibso.chicory.runtime;
+
+import com.dylibso.chicory.wasm.types.Instruction;
+
+@FunctionalInterface
+public interface ExecutionListener {
```

Comment:
Will add 5 cents from my side.

You can use standard functional interface from JDK here, `Consumer` and just create `ExecutionContext` dto to pass all required context.


The code will look like 
```
Consumer<ExecutionContext> listener;
```
and
```
public class ExecutionContext {

    private final Instruction instruction;
    private final long[] operands;

    private final MStack stack;
...
}
```

Reply:
Thanks for chipping in, not super concerned nor fussy about the style personally.
The proposed pattern is already used [here](https://github.com/dylibso/chicory/blob/main/wasm/src/main/java/com/dylibso/chicory/wasm/ParserListener.java) and that's the only reason why I used it.

Reply:
I also would prefer the style suggested by @mstepan. This would also have the advantage of being more flexible, giving the possibility of adding more fields to that `ExecutionContext` without changing any interface signature.